### PR TITLE
Don't discard globals on a COMMIT

### PR DESCRIPTION
**Pull request by msullivan**

### CODE REVIEW

File: tests/test_server_proto.py [hunk 1]
```diff
@@ -517,6 +517,12 @@ async def test_server_proto_set_global_01(self):
 
         await self.con.query('reset global glob')
 
+    async def test_server_proto_set_global_02(self):
+        await self.con.execute('START TRANSACTION')
+        await self.con.execute('set global glob := 1')
```
Comment (elprans) [inline tests/test_server_proto.py:522]: This makes me wonder if we want to support transaction-scoped globals.

**Review by elprans — APPROVED** [review]

File: tests/test_server_proto.py (hunk 1 above)
Comment (msullivan) [reply tests/test_server_proto.py:522]: I think that in practice with the new protocol, this won't matter at all?

Comment (msullivan) [reply tests/test_server_proto.py:522]: Since setting globals will be done on a language binding level and not via `SET GLOBAL`

Comment (elprans) [reply tests/test_server_proto.py:522]: Sure, we can circle back to this when and if this becomes needed at the language level.

**Pull request merged**